### ISSUE TITLE
DNN using iris dataset on spark 

### ISSUE THREAD

**Issue by salman-khan-mrd** [deeplearning4j/deeplearning4j#5440]
public static void main(String[] args) throws  Exception {

    SparkConf conf = new SparkConf();
    conf.setAppName("DataVec Example");
 	JavaSparkContext sc = new JavaSparkContext(conf);
       
    JavaRDD<String> stringData = sc.textFile("hdfs://192.168.26.88:54310/user/hadoop/iris.txt");
    RecordReader rr = new CSVRecordReader();
    JavaRDD<List<Writable>> parsedInputData = stringData.map(new StringToWritablesFunction(rr));
  SplitTestAndTrain testAndTrain =     parsedInputData.splitTestAndTrain(0.65);  //Use 65% of data for training

   DataSet trainingData = testAndTrain.getTrain();
   DataSet testData = testAndTrain.getTest();
              
   DataNormalization normalizer = new NormalizerStandardize();
   normalizer.fit(trainingData);           //Collect the statistics (mean/stdev) from the training data. This does not modify the input data
   normalizer.transform(trainingData);     //Apply normalization to the training data
  normalizer.transform(testData);         //Apply normalization to the test data. This is using statistics calculated from the *training* set
        
     
        
               
        
    //    JavaRDD<DataSet> trainData1 = sc.parallelize(t1);
   //     JavaRDD<DataSet> testData1 = sc.parallelize(t2);
        
 
                
        
        final int numInputs = 4;
        int outputNum = 3;
        int iterations = 100;
        long seed = 1234;

        System.out.print("Build model....");
        
        MultiLayerConfiguration conf1 = new NeuralNetConfiguration.Builder()
            .seed(seed)
            .iterations(iterations)
            .activation(Activation.TANH)
            .weightInit(WeightInit.XAVIER)
            .learningRate(0.06)
            .regularization(true).l2(1e-4)
            .list()
            .layer(0, new DenseLayer.Builder().nIn(numInputs).nOut(3)
                .build())
            .layer(1, new DenseLayer.Builder().nIn(3).nOut(3)
                .build())
            .layer(2, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                .activation(Activation.SOFTMAX)
                .nIn(3).nOut(outputNum).build())
            .backprop(true).pretrain(false)
            .build();

        //run the model
        MultiLayerNetwork model = new MultiLayerNetwork(conf1);
        model.init();
    //  model.setListeners(new ScoreIterationListener(100));

     System.out.println(" Inialization Network ..............");
        
       
     SparkDl4jMultiLayer master = new SparkDl4jMultiLayer(sc,conf1,new ParameterAveragingTrainingMaster(true,2,1,5,1,0));

     MultiLayerNetwork network2 = master.fitLabeledPoint(trainingData);
    
     
     Evaluation evaluation = new Evaluation();
     evaluation.eval(testData.getLabels(), network2.output(testData.getFeatureMatrix()));
     System.out.println(evaluation.stats());

cast problem 
cant run on spark 

@AlexDBlack
@raver120
@treo

**Comment by AlexDBlack** [comment]
As discussed in gitter - the problem is not related to DL4J.
> SplitTestAndTrain testAndTrain = parsedInputData.splitTestAndTrain(0.65)

JavaRDD.splitTestAndTrain method does not exist.

**Comment by salman-khan-mrd** [comment]
can u plz implement the following ur eaxmple on spark culuster ....... add it to spark examples 
we students are learning alot from you but this time we are stuck  
@AlexDBlack
@raver120
@treo
https://github.com/deeplearning4j/dl4j-examples/blob/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/dataexamples/CSVExample.java

**Comment by salman-khan-mrd** [comment]
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.datavec.api.records.reader.RecordReader;
import org.datavec.api.records.reader.impl.csv.CSVRecordReader;
import org.datavec.api.transform.schema.Schema;
import org.deeplearning4j.eval.Evaluation;
import org.deeplearning4j.nn.conf.MultiLayerConfiguration;
import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
import org.deeplearning4j.nn.conf.layers.DenseLayer;
import org.deeplearning4j.nn.conf.layers.OutputLayer;
import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
import org.deeplearning4j.nn.weights.WeightInit;
import org.deeplearning4j.optimize.listeners.ScoreIterationListener;
import org.deeplearning4j.spark.impl.multilayer.SparkDl4jMultiLayer;
import org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingMaster;
import org.nd4j.linalg.activations.Activation;
import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.dataset.DataSet;
import org.nd4j.linalg.dataset.SplitTestAndTrain;
import org.nd4j.linalg.dataset.api.preprocessor.DataNormalization;
import org.nd4j.linalg.dataset.api.preprocessor.NormalizerStandardize;
import org.nd4j.linalg.lossfunctions.LossFunctions;
import org.datavec.api.writable.Writable;
import org.datavec.spark.transform.misc.StringToWritablesFunction;

public class iris {

   

    
	@SuppressWarnings("unchecked")
	public static void main(String[] args) throws  Exception {

    SparkConf conf = new SparkConf();
    conf.setAppName("DataVec Example");
 	JavaSparkContext sc = new JavaSparkContext(conf);

   
    JavaRDD<String> stringData = sc.textFile("hdfs://192.168.26.88:54310/user/hadoop/iris.txt");
    RecordReader rr = new CSVRecordReader();
    JavaRDD<List<Writable>> parsedInputData = stringData.map(new StringToWritablesFunction(rr));
  SplitTestAndTrain testAndTrain =     parsedInputData.splitTestAndTrain(0.65);  //Use 65% of data for training

   DataSet trainingData = testAndTrain.getTrain();
   DataSet testData = testAndTrain.getTest();
              
   DataNormalization normalizer = new NormalizerStandardize();
   normalizer.fit(trainingData);           //Collect the statistics (mean/stdev) from the training data. This does not modify the input data
   normalizer.transform(trainingData);     //Apply normalization to the training data
  normalizer.transform(testData);         //Apply normalization to the test data. This is using statistics calculated from the *training* set
        
     
        
               
        
    //    JavaRDD<DataSet> trainData1 = sc.parallelize(t1);
   //     JavaRDD<DataSet> testData1 = sc.parallelize(t2);
        
 
                
        
        final int numInputs = 4;
        int outputNum = 3;
        int iterations = 100;
        long seed = 1234;

        System.out.print("Build model....");
        
        MultiLayerConfiguration conf1 = new NeuralNetConfiguration.Builder()
            .seed(seed)
            .iterations(iterations)
            .activation(Activation.TANH)
            .weightInit(WeightInit.XAVIER)
            .learningRate(0.06)
            .regularization(true).l2(1e-4)
            .list()
            .layer(0, new DenseLayer.Builder().nIn(numInputs).nOut(3)
                .build())
            .layer(1, new DenseLayer.Builder().nIn(3).nOut(3)
                .build())
            .layer(2, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                .activation(Activation.SOFTMAX)
                .nIn(3).nOut(outputNum).build())
            .backprop(true).pretrain(false)
            .build();

        //run the model
        MultiLayerNetwork model = new MultiLayerNetwork(conf1);
        model.init();
    //  model.setListeners(new ScoreIterationListener(100));

     System.out.println(" Inialization Network ..............");
        
       
     SparkDl4jMultiLayer master = new SparkDl4jMultiLayer(sc,conf1,new ParameterAveragingTrainingMaster(true,2,1,5,1,0));

     MultiLayerNetwork network2 = master.fitLabeledPoint(trainingData);
    
     
     Evaluation evaluation = new Evaluation();
     evaluation.eval(testData.getLabels(), network2.output(testData.getFeatureMatrix()));
     System.out.println(evaluation.stats());

    }
   
}
 error in evaluation
